### PR TITLE
fix: dedupe duplicate dashboard watchlist entries

### LINKED ISSUE #65: bug: dashboard can show duplicate entries when multiple users watchlist the same item
Items that are watchlisted by two users close together can appear duplicated on the dashboard.

Example observed:
- `A Life of a King` is showing twice on the dashboard
- One entry is attributed to one user and the other entry is attributed to another user

However, on the `Watchlists` page the same item is handled correctly and only appears once, with both users shown under the same request.

This suggests the issue is specific to how the dashboard aggregates or renders recently watchlisted items, rather than the underlying watchlist data itself.

Things to investigate:
- Whether the dashboard is grouping entries incorrectly when multiple users watchlist the same item close together
- Whether timing/order of watchlist events affects how duplicates are created
- Whether the dashboard is rendering separate activity events instead of a unified media item view
- Why the `Watchlists` page is deduplicating correctly while the dashboard is not

Expected behaviour:
If multiple users watchlist the same item, the dashboard should present it as a single item with the relevant users associated with it, rather than showing duplicate entries.

### PR DESCRIPTION
**Pull request by Migz93**
Closes #65

## Summary

This PR fixes a dashboard duplication bug where the same watchlisted title could appear twice when different users added it close together. The dashboard now collapses entries that represent the same media even when Plex cached them under different item IDs, so the dashboard matches the grouped behavior already used on the Watchlists page.

## Changes

- Updated the dashboard aggregation in `src/server/db/sync.ts` to merge watchlist rows not only by exact `plex_item_id`, but also by shared GUIDs when the same media was cached under different Plex ID formats.
- Kept the merged dashboard cards user-aware by preserving the combined user list, the most recent watchlist timestamp, and availability state without duplicating the same user on a card.
- Updated dashboard movie/show counts to use the deduplicated dashboard item set so the summary stats stay consistent with the visible cards.
- Added `tests/server/dashboard.test.ts` as a regression test covering the exact duplicate scenario from the bug report: two users, one title, different cached Plex IDs, shared GUIDs, and one expected dashboard item.

## Test plan

- [x] Run `node --test --import tsx tests/server/dashboard.test.ts`
- [x] Run `npm run check`
- [x] Run `npm run build`
- [x] Rebuild the live `hubarr` Docker container from this branch and verify the duplicate dashboard entry is gone in the local live instance

🤖 Generated with Codex

### CODE REVIEW

**Comment by Migz93** [comment]
## Code Review

The fix is correct for the common case and the test covers the specific bug scenario well. Two things to flag:

### Bug: Transitive merges are silently dropped

The merge logic only handles one hop. If three distinct `plex_item_id` values represent the same media in a chain — e.g. item A and B share `imdb://tt123`, item B and C share `tmdb://456` — the resolution breaks:

1. `guidToCanonical` gets set to `B` for `tmdb://456`
2. `mergeInto` maps `A → B` and `C → B`
3. In the merge loop, A is merged into B and **B is deleted** from `grouped`
4. When C tries to merge into B, `grouped.get(B)` returns `undefined` → `continue` → **C is left as a separate card**

The same chain problem occurs regardless of iteration order. Three different Plex IDs for one movie is rare but the kind of edge case that will eventually surface.

A straightforward fix would be to resolve each merge target through the chain to its final canonical before executing merges — e.g. follow `mergeInto` pointers until you reach an item not in `mergeInto`, and rewrite all intermediate entries to point directly at that root.

### Minor: GUIDs from the existing-key branch are ignored

When `grouped.get(row.plexItemId)` finds an existing entry, `rawPayload` is never parsed and GUIDs are never updated. This is fine because GUIDs for a given `plexItemId` should be identical across rows — but if they ever diverge (stale cache, partial resync), only the first row's GUIDs would be used. A short comment here would save a future reader from wondering whether this was intentional.

### Test looks good

Covers the exact regression scenario (two users, one title, different cached Plex IDs, shared GUIDs) and asserts both the card count and the stats. No issues there.

🤖 Generated with Claude Code

File: src/server/db/sync.ts [hunk 1]
```diff
@@ -206,30 +213,59 @@ export function buildDashboard(db: Database.Database): DashboardResponse {
...
+  const guidToCanonical = new Map<string, string>();
+  const mergeInto = new Map<string, string>();
+
+  for (const [plexItemId, guids] of itemGuids) {
+    for (const guid of guids) {
+      if (guidToCanonical.has(guid)) {
+        const canonical = guidToCanonical.get(guid)!;
+        if (canonical !== plexItemId && !mergeInto.has(plexItemId)) {
+          mergeInto.set(plexItemId, canonical);
+        }
```
[the remark this answers is not in the document]
Comment (Migz93) [reply src/server/db/sync.ts]: Addressed in `41fdbbb`.

The merge planner now resolves transitive chains to their final canonical target before applying merges, so `A <- B <- C` style overlaps collapse into a single item instead of leaving the last link behind. I added regression tests for that chain case in both the dashboard and grouped watchlist paths.

🤖 Generated with Codex

File: src/server/db/sync.ts [hunk 2]
```diff
@@ -206,30 +213,59 @@ export function buildDashboard(db: Database.Database): DashboardResponse {
...
+      } else {
+        guidToCanonical.set(guid, plexItemId);
+      }
+    }
+  }
+
+  for (const [sourceId, targetId] of mergeInto) {
+    const source = grouped.get(sourceId);
+    const target = grouped.get(targetId);
+    if (!source || !target) continue;
```
[the remark this answers is not in the document]
Comment (Migz93) [reply src/server/db/sync.ts:229]: Addressed in `41fdbbb`.

The GUID merge planner now scopes matches by media type before considering items equivalent, so movie/show rows that happen to share a provider GUID no longer collapse into one dashboard or watchlist item. I also added regression coverage for the cross-type case in both dashboard and grouped watchlist tests.

🤖 Generated with Codex

**Comment by Migz93** [comment]
Addressed the review feedback in `41fdbbb`.

This follow-up extracts the GUID merge planner so the dashboard and grouped watchlist paths use the same logic, resolves transitive GUID chains before merging, scopes GUID matches by media type to prevent movie/show collisions, and adds regression tests for both edge cases.

🤖 Generated with Codex
